### PR TITLE
Update component documentation to match package name conventions

### PR DESCRIPTION
**Pull request by dustmop**
### What does this PR do?

Update documentation on how to create components such that it matches the conventions used for package names in new style components.

### Motivation

Keeping docs consistent with linter rules from https://github.com/DataDog/datadog-agent/pull/27868

### Describe how to test/QA your changes

N/A

### CODE REVIEW

File: docs/public/components/creating-components.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ In the example of a compression component, the def folder looks like this:
     }
     ```
 
-All component interfaces must be called `Component`, so all imports have the form `def.Component`.
+All component interfaces must be called `Component`, so all imports have the form `<component>.Component`.
```
Comment (neko-dd) [inline docs/public/components/creating-components.md:83]: Assuming this is a placeholder,

```suggestion
All component interfaces must be called `Component`, so all imports have the form `<COMPONENT_NAME>.Component`.
```

File: docs/public/components/creating-components.md [hunk 2]
```diff
@@ -114,11 +114,11 @@ internally by each component (more on this [here TODO]()). -->
 ### The impl folders
 
 The `impl` folder is where the component implementation is written. The details of component implementation are up to the developer.
-The only requirement is that there is a public instantiation function called `NewComponent`.
+The only requirements are that the package name follows the pattern `<component>impl` and that there is a public instantiation function called `NewComponent`.
```
Comment (neko-dd) [inline docs/public/components/creating-components.md:117]:
```suggestion
The only requirements are that the package name follows the pattern `<COMPONENT_NAME>impl` and that there is a public instantiation function called `NewComponent`.
```

**Review by neko-dd — APPROVED** [review]
2 small changes but otherwise LGTM

**Review by jeremy-hanna — APPROVED** [review]

**Pull request merged**